CDCgov/prime-reportstream
# Experience/refactor validate for reuse [6092]

This change splits out the existing `Validate` page into a number of reusable components. The goal of this change is to allow for creation of a new `Upload` page, very similar in functionality and appearance to the existing`Validate` page, without duplication of code and functionality.

In so doing, the assumptions are made that the `Upload` and `Validate` pages will be identical beyond the content and functionality changes governed by the props made available to `FileHandler`:

* fetch function to be called when uploading files
* various pieces of text:
    * page heading
    * message shown on successful upload
    * label for upload form
    * text for reset button
    * text for submit button

Note that this includes a refactor to the Waters API for type checking purposes, in order to make sharing functionality easier here. There are no functional changes there.

## Test Steps:

No functional or design changes here, this is a pure refactor of code. You can use the test steps from https://github.com/CDCgov/prime-reportstream/pull/6079 to make sure everything is still working as expected


## Changes
* splits validation success and error displays into individual components
* splits validation submit button into individual component
* splits validation form into individual component
* splits remainder of `Validate` into a reusable `FileHandler` component that can be controlled or customized via props
* refactor of Waters API for type checking

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
